uaclient/conftest.py

```diff
@@ -0,0 +1,41 @@
+import io
+import logging
+
+import pytest
+
+
+@pytest.fixture
+def caplog_text(request):
```

Comment:
This function works around missing functionality in the versions of pytest in trusty and xenial; I recommend skipping over it on an initial review and coming back to it once you see how it's used.

(Note that in xenial we will be able to replace this with `pytest-catchlog` which is packaged in Ubuntu as `python3-pytest-catchlog`; that project was integrated in to pytest between xenial and bionic.)